include/mqtt/endpoint.hpp

```diff
@@ -1466,17 +1470,14 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
      *        Properties<BR>
      *        See https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901145<BR>
      *        3.6.2.2 PUBREL Properties
-     * @param life_keeper
-     *        An object that stays alive (but is moved with force_move()) until the async operation is finished.
      * @param
      */
     void pubrel(
         packet_id_t packet_id,
         v5::pubrel_reason_code reason_code = v5::pubrel_reason_code::success,
-        v5::properties props = {},
-        any life_keeper = any()
+        v5::properties props = {}
```

Comment:
You removed life_keeper here. I agree with the removal. But I'd like to clarify the following point.

* `props` must manage their lifetime by themselves.

`v5::properties` has `v5::property_variant`. The actual type of `v5::property_variant` could be a string property such as `user_property`. The string property has `buffer`. The `buffer` has `const_shared_ptr_array`. It can be null.

If user create intentionally the reference type of `buffer` and pass it to the string property, then pass the string property to pubrel() as the part of the props, then the (removed) parameter life_keeper can be used to keep the lifetime of the `buffer`. But I think that it is unexpected usage. It is a kind of abuse. (But not documented well)

When properties are received by `mqtt_cpp`, then the properties might be share their lifetime. Each `buffer`'s `const_shared_ptr_array` are the same. In this case, there is no problem to pass to the properties as the parameter of pubrel().